R/meta.R

```diff
@@ -0,0 +1,228 @@
+    ),
+    class = 'tplyr_meta'
+  )
+}
+
+#' Add variables to a tplyr_meta object
+#'
+#' Add additional variable names to a `tplyr_meta()` object.
+#'
+#' @param meta A tplyr_meta object
+#' @param names A list of names, providing variable names of interest. Provide
+#'   as a list of quosures using `rlang::quos()`
+#'
+#' @return tplyr_meta object
+#' @md
+#'
+#' @family Metadata additions
+#' @rdname metadata_additions
+#'
+#' @export
+#'
+#' @examples
+#'
+#' m <- tplyr_meta()
+#' m <- add_variables(m, quos(a, b, c))
+#' m <- add_filters(m, quos(a==1, b==2, c==3))
+#' m
+add_variables <- function(meta, names) {
+
+  if (!all(map_lgl(names, ~ is_quosure(.) && quo_is_symbol(.)))) {
+    stop("Names must be provided as a list of names", call.=FALSE)
+  }
+
+  if (!inherits(meta, 'tplyr_meta')) {
+    stop("meta must be a tplyr_meta object", call.=FALSE)
+  }
+
+  add_variables_(meta, names)
+}
+
+#' Internal application of variables onto tplyr_meta object
+#' @noRd
+add_variables_ <- function(meta, names) {
+  meta$names <- append(meta$names, names)
+  meta
+}
+
+#' @param filters A list of symbols, providing variable names of interest. Provide
+#'   as a list of quosures using `rlang::quos()`
+#'
+#' @family Metadata additions
+#' @rdname metadata_additions
+#'
+#' @export
+add_filters <- function(meta, filters) {
+
+  if (!all(map_lgl(filters, ~ is_quosure(.) && quo_is_call(.)))) {
+    stop("Filters must be provided as a list of calls", call.=FALSE)
+  }
+
+  if (!inherits(meta, 'tplyr_meta')) {
+    stop("meta must be a tplyr_meta object", call.=FALSE)
+  }
+
+  add_filters_(meta, filters)
+}
+
+#' Internal application of filters onto tplyr_meta object
+#' @noRd
+add_filters_ <- function(meta, filters) {
+  meta$filters <- append(meta$filters, filters)
+  meta
+}
+
+#' Get the metadata dataframe from a tplyr_table
+#'
+#' Pull out the metadata dataframe from a tplyr_table to work with it directly
+#'
+#' @param t
+#'
+#' @return Tplyr metadata dataframe
+#' @export
+#'
+#' @examples
+#' t <- tplyr_table(mtcars, gear) %>%
+#'   add_layer(
+#'     group_desc(wt)
+#'   )
+#'
+#' t %>%
+#'   build(metadata=TRUE)
+#'
+#' get_metadata(t)
+get_metadata <- function(t) {
+
+  if (!inherits(t, 'tplyr_table')) {
+    stop("t must be a tplyr_table object", call.=FALSE)
+  }
+
+  if (is.null(t$metadata)){
+    stop(paste(
+      "t does not contain a metadata dataframe.",
+      "Make sure the tplyr_table was built with `build(metadata=TRUE)`"))
+  }
+
+  return(t$metadata)
+}
+
+#' Append the Tplyr table metadata dataframe
+#'
+#' `append_metadata()` allows a user to extend the Tplyr metadata data frame
+#' with user provided data. In some tables, Tplyr may be able to provided most
+#' of the data, but a user may have to extend the table with other summaries,
+#' statistics, etc. This function allows the user to extend the tplyr_table's
+#' metadata with their own metadata content using custom data frames created
+#' using the tplyr_meta() object.
```

Comment:
tplyr_meta is a class, not a function. So I'd remove the parenthesis. Additionally, add back ticks so rendered help file is formatted.

Reply:
> * The argument 'target' is documented for `get_meta_subset` but not in the function signature. Fix is to remove `@param target ...` from the roxygen comment.

This is a bit of a weird doc situation - it's not in the S3 general method but it's in the data.frame classed version.

Reply:
^^^ this was a stupid comment by me. Fixing.